### ISSUE TITLE
How to use DownGit with wget or curl?

### ISSUE THREAD

**Issue by sugatoray** [MinhasKamal/DownGit#42]
How to use DownGit generated URL with `wget` or `curl`? Since the url does not have any file extension, it creates an `index.html` file. Also, is there any way that I can use DownGit generated URL in python to download the zipped file?

**Comment by MinhasKamal** [comment]
If you want to use `wget`, `curl`, or `python`, then you don't need DownGit. You can directly call GitHub API and download the files.

**Comment by sugatoray** [comment]
> If you want to use `wget`, `curl`, or `python`, then you don't need DownGit. You can directly call GitHub API and download the files.

@MinhasKamal Could you please share a code snippet for downloading a subdirectory of a github repo? Thank you.

**Comment by MinhasKamal** [comment]
@sugatoray The code for downloading GitHub directories should be pretty straight-forward (I have never tried though)- 
1. take the GutHub directory link 
2. hit the GitHub API (manual- https://docs.github.com/en/rest)
3. search through the directory and download all files
 
You will find many examples by Googling. Here is a python code I found- https://gist.github.com/pdashford/2e4bcd4fc2343e2fd03efe4da17f577d

**Comment by sugatoray** [comment]
@MinhasKamal Thank you, for the examples. Much appreciated!

**Comment by sugatoray** [comment]
@MinhasKamal Could you please activate the "discussion" tab, so if there are questions (which are not exactly issues), people can ask them under discussions? This will also keep the number of open issues low (hopefully).

I have another question for you: *Is the following link also equivalent to appending a .zip at the end?*

```sh
https://minhaskamal.github.io/DownGit/#/home?url=https://github.com/sugatoray/latex-project-report-template
```

I was trying to use downgit to open any folder in overleaf.com using their api. I know that this works.

https://www.overleaf.com/docs?snip_uri=https://github.com/sugatoray/latex-project-report-template/archive/refs/heads/master.zip

API: https://www.overleaf.com/dev

However, I would like to use the path to any folder (not the just the entire branch). Any suggestions?

**Comment by MinhasKamal** [comment]
No, you cannot use DownGit for this purpose. https://github.com/sugatoray/latex-project-report-template/archive/refs/heads/master.zip is a direct file/resource link. For your purpose, you have to use  DownGit, or implement something like it on a server-side. 

I am actually an embedded-system developer, so my knowledge of web-app is very limited.

**Comment by sugatoray** [comment]
@MinhasKamal Thank you, for the quick response.

**Comment by suntong** [comment]
> If you want to use `wget`, `curl`, or `python`, then you don't need DownGit. You can directly call GitHub API and download the files.

Hi MinhasKamal, would you explain a bit how the download link work please? I.e., since you're providing a download link functionality, why disabling  `wget`, `curl` downloading? (I took a look at your suggested GitHub API & python code, they are much more complicated than using `wget`, `curl` to download from an url)

For my case, I don’t want to download to my local drive but need a download link to do the download in my remote VPS instead.
